tests/test_models.py

```diff
@@ -280,3 +284,105 @@ class Cart(BaseModel):
 
     name_field: serializers.Field = items_field.child.fields["name"]
     assert isinstance(name_field, serializers.CharField)
+
+
+def test_enum_model():
+    class CountryEnum(Enum):
+        US = 'US'
+        GB = 'GB'
+        FR = 'FR'
+
+    class NotificationPreferenceEnum(Enum):
+        NONE = 'no_notifications'
+        SOME = 'some_notifications'
+        ALL = 'all_notifications'
+
+    class Person(BaseModel):
+        name: str
+        email: pydantic.EmailStr
+        age: int
+        height: float
+        date_of_birth: datetime.date
+        notification_preferences: NotificationPreferenceEnum
+        original_nationality: typing.Optional[CountryEnum]
+        nationality: CountryEnum = CountryEnum.GB
+
+    serializer = Person.drf_serializer()
+
+    assert serializer.__class__.__name__ == "PersonSerializer"
+    assert len(serializer.fields) == 8
+
+    # Regular fields
+    assert isinstance(serializer.fields["name"], serializers.CharField)
+    assert isinstance(serializer.fields["email"], serializers.EmailField)
+    assert isinstance(serializer.fields["age"], serializers.IntegerField)
+    assert isinstance(serializer.fields["height"], serializers.FloatField)
+    assert isinstance(serializer.fields["date_of_birth"], serializers.DateField)
+    assert isinstance(serializer.fields["notification_preferences"], EnumField)
+    for name in [
+        "name",
+        "email",
+        "age",
+        "height",
+        "date_of_birth",
+        "notification_preferences"
+    ]:
+        field = serializer.fields[name]
+        assert field.required is True, name
+        assert field.default is serializers.empty, name
+        assert field.allow_null is False, name
+        if name == 'notification_preferences':
+            assert field.choices == dict(
+                [(x, x.name) for x in NotificationPreferenceEnum]
+            )
+
+    # Optional
+    field: serializers.Field = serializer.fields["original_nationality"]
+    assert isinstance(field, EnumField)
+    assert field.allow_null is True
+    assert field.default is None
+    assert field.required is False
+    assert field.choices == dict([(x, x.name) for x in CountryEnum])
+
+    # With default
+    field: serializers.Field = serializer.fields["nationality"]
+    assert isinstance(field, EnumField)
+    assert field.allow_null is False
+    assert field.default == CountryEnum.GB
+    assert field.required is False
+    assert field.choices == dict([(x, x.name) for x in CountryEnum])
+
+
+@patch(
+    'drf_pydantic.fields.EnumField.fail',
+    side_effect=ValidationError("bad_value is not a valid choice.")
+)
+def test_enum_value(fail):
```

Comment:
Are you testing an error here (I'm not familiar with `unittest`)? If so, I suggest using `with pytest.raises` as seen here https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions to stay consistent with other tests

Reply:
It was yes because I'd been having issues with the fail method wanting Django configured, however that had already been fixed so this isn't needed at all anymore and has been removed.